lib/mongo/client_encryption.rb

```diff
@@ -71,9 +99,53 @@ def create_data_key
       end
 
       data_key_document = Hash.from_bson(BSON::ByteBuffer.new(result))
-      insert_result = @client.use(@key_vault_db_name)[@key_vault_coll_name].insert_one(data_key_document)
+      insert_result = @collection.insert_one(data_key_document)
+
+      return insert_result.inserted_id.data
+    end
+
+    # Encrypts a value using the specified encryption key and algorithm
+    #
+    # @param [ String|Numeric ] value The value to encrypt
+    # @param [ Hash ] opts
+    #
+    # @option [ String ] :key_id The UUID of the encryption key as
+    #   it is stored in the key vault collection
+    # @option [ String ] :algorithm The algorithm used to encrypt the value.
+    #   Valid algorithms are "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+    #   or "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+    #
+    # @return [ String ] The encrypted value
+    #
+    # This method is not currently unit tested.
+    # Find tests in spec/integration/explicit_encryption_spec.rb
+    def encrypt(value, opts={})
+      result = nil
+      value = { 'v': value }.to_bson.to_s
+
+      Crypt::ExplicitEncryptionContext.with_context(@crypt_handle.ref, @io, value, opts) do |context|
+        result = context.run_state_machine
```

Comment:
`with_context` should return the value of the block, and I believe it already works this way, making explicit assignment to result unnecessary as far as I can tell.